### PR TITLE
DEV-302: use automatic backups for creation of acceptance RDS instance

### PR DESCRIPTION
**Pull request by xaniasd**
- Stop creating manual snapshots of dbs3 for acceptance RDS instances and use the latest automatic backup
- Disable backup retention for acceptance databases
- Remove redundant user data backup

### CODE REVIEW

File: acceptance-amazon-rds-sync/rds-common [hunk 1]
```diff
@@ -7,10 +7,30 @@ REGEX_PROGRESS="<PercentProgress>([0-9]+)</PercentProgress>"
...
+	SNAPSHOTS=$(rds-describe-db-snapshots -i "$SNAPSHOT_INSTANCE");
+	current_date=$(date +%Y-%m-%d);
+
+	while read -r line; do
+		stringarray=($line);
+		snapshot_name=${stringarray[1]};
+	  # automatic backups named as rds:dbs3-2015-03-02-01-11
+	  if [[ "${snapshot_name}" =~ $REGEX_SNAPSHOT_CREATION_DATE ]]; then
+	  	creation_date=$(date -d ${BASH_REMATCH[1]} +%Y-%m-%d)
+	  	[[ $current_date == $creation_date ]] && latest_snapshot=$snapshot_name;
```
Comment (jaysphoto) [inline acceptance-amazon-rds-sync/rds-common]: Shall we just always use 'the latest' snapshot that is available, regardless of what it's date is ? It should be fine to let the cloud toolkit use whatever is available, and not create dbs3 snapshots at all. That would put the whole management of dbs3 snapshots into AWS's domain and out of the cloud toolkit :-)

File: acceptance-amazon-rds-sync/rds-common [hunk 2]
```diff
@@ -11,6 +11,34 @@ REGEX_DATE="<SnapshotCreateTime>(.*?)T(.*?)Z</SnapshotCreateTime>"
...
+    REGEX="rds:dbs3-[0-9]{4}-[0-9]{2}-[0-9]{2}"
+    if ! [[ "$snapshot_name" =~ $REGEX ]]; then
+    	echo "No automatic backup, $snapshot_name"
+    	continue
+    fi
+
+    # naively find latest creation date
+    if [[ -z $latest_snapshot ]]; then
+    	latest_snapshot=$snapshot_name
+    elif [[ "$snapshot_name" > "$latest_snapshot" ]]; then
```
Comment (jaysphoto) [inline acceptance-amazon-rds-sync/rds-common]: use '-gt' to keep in coding style, not sure if '>' even works in bash ..

Comment (xaniasd) [inline acceptance-amazon-rds-sync/rds-common]: actually we were using -gt, but didn't work for string comparison :santa:

Comment (jaysphoto) [inline acceptance-amazon-rds-sync/rds-common]: ah.. okay fair enough :)

Comment (jaysphoto) [inline acceptance-amazon-rds-sync/rds-common]: :octocat:

Comment (jaysphoto) [inline acceptance-amazon-rds-sync/rds-common]: ps.: please remove the unused SNAPSHOT 'constant' from [config.template](acceptance-amazon-rds-sync/config.template) and update the README.md accordingly ?

Comment (xaniasd) [inline acceptance-amazon-rds-sync/rds-common]: I got an octocat already :P

Comment (jaysphoto) [inline acceptance-amazon-rds-sync/rds-common]: What octocat ? :package:

**Comment by jaysphoto** [comment]
:octocat:

**Comment by silvanu** [comment]
:octocat:

**Pull request merged**